[댓글 기능] 이지 미션 제출합니다

skygl opened the pull request:
마감 기한 때문에 기능 구현을 우선으로 작성해서, 테스트 코드들이 많이 깨지고 테스트 코드도 많이 부족합니다 ㅠ
최대한 테스트 코드를 빠르게 추가, 수정해서 다시 PR을 보내겠습니다
프로덕션 코드를 우선으로 피드백 남겨주시면 감사할것같아요

kimjungmin-developer committed:
[DOCS] README.md 업데이트

kimjungmin-developer committed:
[FEAT] User의 PK를 Id로 교체

kimjungmin-developer committed:
[FIX] 드랍다운 버그 수정

kimjungmin-developer committed:
[FEAT] 작성자만 글 수정가능하게 제한

kimjungmin-developer committed:
[FEAT] 댓글 생성 조회 기능 추가

kimjungmin-developer committed:
[FEAT] 댓글 수정, 삭제 버튼 추가

kimjungmin-developer committed:
[FEAT]댓글 업데이트 기능 추가

kimjungmin-developer committed:
[FEAT] 부모 엔티티 삭제시 자식 엔티티 삭제 기능 추가

kimjungmin-developer committed:
[FIX] 로그아웃 버그 수정

kimjungmin-developer committed:
[FEAT] 댓글 테스트 추가

Commit:
[REFACTOR] mypage edit 수정, 필요없는 파일 삭제

Commit:
[REFACTOR] ArticleController 기존 테스트 에러 수정

phs1116 reviewed (changes requested):
안녕하세요! 리뷰어 화투입니다.
짧은 시간 내에 구현하시느라 정말 고생 많으셨습니다 👍 
몇가지 피드백 추가했는데요, 참고 부탁드립니다. 
테스트 부분에 대해서는 말씀주신대로 PR 주시면 이어서 피드백 드리겠습니다.

Commit:
[REFACTOR] Mockito 사용해서 UserService Test 수정

Commit:
[TEST] ArticleService 테스트 코드 작성

Commit:
[TEST] CommentService 테스트 코드 작성

- ArticleServiceTests 에서 UserRepository를 Mock으로 생성하던 것을 UserService를 Mock으로 생성하는 것으로 변경
- 서비스가 Session에서 로그인 정보를 빼오는 것을 컨트롤러에서 수행하도록 변경

Commit:
[TEST] CommentController 테스트 코드 작성

Commit:
[REFACTOR] DTO 변경

- DTO에서 Entity를 필드로 가지지 않도록 수정
- Converter 클래스를 인터페이스로 추출
- UserDto를 ResponseDto와 RequestDto로 구분
- Converter에서 DTO의 타입을 다르게 하기위해 제너릭을 3개 사용

Commit:
[REFACTOR] Test 코드 에러 수정

Commit:
[FEAT] 비로그인시, 혹은 로그인시 접속 가능한 페이지 처리

Commit:
[FEAT] 댓글의 작성자만 수정, 삭제를 할 수 있는 기능 추가

Commit:
[REFACTOR] 테스트 에러 수정

Commit:
[REFACTOR] Session에서 Email 대신 User를 가지도록 수정

Commit:
[REFACTOR] 공통되는 부분 ControllerTests의 추상 메서드로 추출

phs1116 reviewed (changes requested):
안녕하세요! 리뷰어 화투입니다.
피드백 반영 감사합니다. :)
전체적인 구현 👍 
다만 테스트 및 몇가지 개선하면 좋을 점에 대해서 피드백 추가했습니다.
참고 부탁드리고 궁금하신 점 있으시면 연락주세요. :)

Commit:
[REFACTOR] Converter를 삭제하고 Assembler로 변경

- DTO와 Entity를 서로 바꿔주는 Converter 삭제
- Entity에서 DTO를 만들어주는 Assembler 생성
- DTO와 Entity 생성자 변경, Entity Builder 패턴 적용

Commit:
[REFACTOR] Comment 객체가 작성자인지 판단하도록 리팩토링

- Comment에게 User를 보내 작성자인지 확인하도록 수정
- Comment 수정 시에 Domain에서 작성자 판단 및 내용을 수정하도록 변경

Commit:
[REFACTOR] Entity도 Mock을 사용하도록 리팩토링

Commit:
[FEAT] Article 수정, 삭제에서도 작성자만 가능하도록 수정

- 세션의 값과 비교해서 작성자인지 확인하도록 하는 기능 추가

Commit:
[TEST] Controller, Domain 테스트 추가

- Interceptor에서 작업이 모두 처리되기 전에 먼저 redirect 되는 에러 수정

phs1116 reviewed (approved):
안녕하세요! 리뷰어 화투입니다.
이번 단계 머지하도록 하겠습니다.
지금까지 구현하시느라 수고 많으셨습니다. 👍
